### PR TITLE
(Berlin) Environment Pipeline

### PR DESCRIPTION
**Pull request by Acentelles**
https://hackmd.io/iG-uj5t_T3i82dxBRb_PSA?view#Proposal-The-Berlin-Pipeline

### CODE REVIEW

File: library/BerlinPipeline/src/Juvix/BerlinPipeline/Step/Types.hs [hunk 1]
```diff
@@ -4,45 +4,40 @@ module Juvix.BerlinPipeline.Step.Types
...
+data WorkingEnv = WorkingEnv
+  { currentExp :: [Env],
+    context :: Context.T Sexp.T Sexp.T Sexp.T
+  }
 
-class StepMetaInfo a where
-  name :: Maybe NameSymbol.T
-  meta :: Meta.T
+data ComputationalInput = ComputationalInput
+  { languageData :: WorkingEnv
```
Comment (mariari) [inline library/BerlinPipeline/src/Juvix/BerlinPipeline/Step/Types.hs:34]: These best belong to the `Piepline` module rather than the `Step` one.

**Comment by cyberglot** [comment]
Wait, this is confusing to me.

File: library/BerlinPipeline/src/Juvix/BerlinPipeline/Meta.hs [hunk 2]
```diff
@@ -17,26 +23,18 @@ data T = Meta
   deriving (Eq, Show)
 
 data Success a = Success
-  { meta :: T,
-    result :: a
+  { sMeta :: T,
+    sResult :: a
```
Comment (cyberglot) [inline library/BerlinPipeline/src/Juvix/BerlinPipeline/Meta.hs]: I think we should keep the same names and use `DuplicateRecordFields` pragma, that way the interface becomes clearer.

**Comment by cyberglot** [comment]
One tiny thing, otherwise LGTM.

**Review by cyberglot — APPROVED** [review]

**Pull request merged**